dls_dev_env.sh

```diff
@@ -1,19 +1,23 @@
 #!/bin/bash
 
 # controls_dev sets pip up to look at a local pypi server, which is incomplete
+
+echo "Note: The current diamond pipenv installation is unable to lock with python 3.10."
+echo "      If you want to lock then install your own recent pipenv version"
```

Comment:
We could, it would require setting up a .venv or using pip install pipenv --user (home directory interaction) just for the new pipenv version. I think at the time I thought it wouldn't be a good idea if we were migrating to pip soon anyway - but I can add now

Reply:
Actually nvm, this is a good idea. I've put this script in the latest commit